docs/elements/reference/common.mdx

```diff
@@ -0,0 +1,444 @@
+| `name` | `string` (optional) | Used to target a specific field by name when rendering outside of a `<Field>` component. Default: Automatically inferred |
+| `children` | `({ message: string, code: string }) => React.ReactNode` (optional) | Provide a function as children and access the error's `message` and `code`. |
+
+### Examples {{ toc: false }}
+
+#### Basic
+
+If the `<Field>` is in an error state, `<FieldError>` will display its message.
+
+```tsx
+<Clerk.Field name="unique-name">
+  <Clerk.FieldError />
+</Clerk.Field>
+```
+
+#### Function as children
+
+By having access to both `message` and `code` you can enrich the incoming `message` or localize it by checking for specific `code`.
+
+```tsx
+<Clerk.Field name="unique-name">
+  <Clerk.FieldError>
+    {({ message, code }) => (
+      <span>
+        {message} ({code})
+      </span>
+    )}
+  </Clerk.FieldError>
+</Clerk.Field>
+```
+
+## `<FieldState>`
+
+Programmatically access additional information of the wrapping `<Field>`. Useful for implementing animations when direct access to the state value is necessary. By default, you'll have access to `state`.
+
+If you use `<Input type="password" validatePassword>` additional information in the form of `message` and `codes` is provided.
+
+### Properties {{ toc: false }}
+
+| Name       | Type                                               | Description                                         |
+| ---------- | -------------------------------------------------- | --------------------------------------------------- |
+| `children` | `(args: { state: 'success' \| 'error' \| 'warning' \| 'info' \| 'idle'; message: string \| undefined; codes: ErrorMessageKey[] \| undefined }) => React.ReactNode` | Use this function to access the field's state. Optionally, information regarding password validation is given. |
+
+### Examples {{ toc: false }}
+
+#### Basic
+
+Initially, the `<Input>` will have a `state` of `idle` until the user interacts with the input. Depending on the user's input, the `state` will change to `success` or `error`.
+
+```tsx {4-6}
+<Clerk.Field name="emailAddress">
+  <Clerk.Label>Email</Clerk.Label>
+  <Clerk.Input />
+  <Clerk.FieldState>
+    {({ state }) => <p>Field's state is: {state}</p>}
+  </Clerk.FieldState>
+</Clerk.Field>
+```
+
+#### `<Input type="password" validatePassword>`
+
+If you're using [`<Input type="password" validatePassword>`](#input-type-password) the `<FieldState>`'s children function receives additional arguments: `codes` and `message`.
+
+Initially, the `<Input>` will have a `state` of `idle` until the user interacts with the input. Depending on the user's input, the `state` will change to `'success' | 'error' | 'warning' | 'info'`.
+
+`message` is the english, standardized message generated for the current state of the input. The `message` is generated through `codes`. By having access to `codes` yourself you can customize or localize the returned `message`.
```

Comment:
```suggestion
If you're using [`<Input type="password" validatePassword>`](#input-type-password) the `<FieldState>`'s children function receives additional arguments:

- `message` – The standardized English message generated for the current state of the input. This message is generated based on the codes associated with the `<FieldState>`.
- `codes` –  The codes associated with the `<FieldState>`. You can use these codes to return custom `message` or localize its contents.

Initially, the `<Input>` will have a `state` of `idle` until the user interacts with the input. Depending on the user's input, the `state` will change to `'success' | 'error' | 'warning' | 'info'`.
```

Swapped paragraphs and some words around and tried to add a slightly clearer explanation of the relationship between `codes` and `message`. Feel like I got it wrong though. Is my change accurate?